text/metrics/0238-pipeline-monitoring.md

```diff
@@ -0,0 +1,426 @@
+the pipeline should match the number of items successfully received,
+otherwise the system is capable of reporting the combined losses to
+the user.
+
+When losses are unacceptable to the user, or the causes of loss cannot
+be resolved through other system indicators, the user (or the system
+acting on the user's behalf) may wish to enable normal-level detail
+for SDKs or enable metrics for intermediate collectors.  The
+additional detail will give the system information that can be used to
+narrow down the location(s) and source(s) where loss occurs.
+
+## OpenTelemetry SDKs have no receivers
+
+OpenTelemetry SDKs are special pipeline components because they do not
+receive data from an external source.  OpenTelemetry SDKs support
+specific, signal-specific features that may appear like standard
+receivers or processors in a pipeline, but as specified here, pipeline
+monitoring only applies to items of data that are submitted for
+export by an SDK-internal mechanism.
+
+There are several examples:
+
+- In the trace SDK, spans are sampled by a component that emulates a
+  pipeline processor, but the actions of a sampler are not the same as
+  an export pipeline.  While we could count spans not sampled using
+  `otel.sdk.dropped` with `success=true` and
+  `name=traceidratiosampler`, it could lead to misleading
+  interpretation because spans that are unfinished are neither
+  `success=true` nor `success=false`.  Different semantic conventions
+  should probably be used to monitor sampler components.
+- In the metrics SDK, Metric View configuration can cause metric
+  events to be "dropped".  We could count all metric data points as
+  logically entering a pipeline, and then the ones that are dropped
+  would appear as `otel.sdk.dropped` with `success=true` and
+  `name=metricreader`, but this would lead to an accounting problem.
+  The point of Drop aggregation is to avoid the cost of a metric
+  instrument, so we do not which to count what we drop.
+
+For these reasons, the `otel.sdk.received` metric is defined as the
+number of items that the SDK produces as input to the pipeline.  This
+quantity MUST be the number of items that are are expected for
+delivery at the final destination, when the pipeline is operating
+correctly and without failures.
+
+## Metrics SDK special considerations
+
+We expect that Metrics SDKs will be used to generate
+pipeline-monitoring metrics reporting about themselves.
+
+As stated above, SDKs SHOULD support configuring an alternate Meter
+Provider for pipeline-monitoring metrics.  When the global Meter
+Provider is used, the Metrics SDK's pipeline will receive its own
+pipeline-monitoring metrics.  When a custom Meter Provider is used, a
+secondary pipeline will receive the pipeline monitoring metrics, in
+which case the secondary pipeline may also self-report for itself.
+
+## Trade-offs and mitigations
+
+The use of three-levels of metric detail may seem like more freedom
```

Comment:
I buy the user experience you want to provide, but I wonder if creating a solution just for the sdk/collector is the correct scope.  For example, someone from gRPC recently asked how to provide a small set of metrics by default, but allow enabling more detailed metrics.  Should this functionality be part of the metrics API/SDK?  Otherwise, I suspect many instrumentation libraries will build their own, slightly different solutions to this problem.  I'm also not a huge fan of having two places (views and detail) to configure which attributes to keep/drop.

Reply:
The concept of additional detail seems to be unwelcome. I will remove before re-opening.